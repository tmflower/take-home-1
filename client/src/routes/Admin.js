// import { useState } from "react";
import { Box, Typography, Table, TableHead, TableBody, TableRow, TableCell } from "@mui/material";
import theme from "../theme";

export function Admin({users}) {

    return (
        <Box sx={{margin: '8rem'}}>
            <Typography variant="h2" sx={{margin: '6rem 0 0 0', display: 'flex', flexDirection: 'column', alignItems: 'center'}}>Admin</Typography>
            <Table>
                <TableHead>
                    <TableRow>
                        <TableCell sx={{color: theme.typography.primary.color, fontSize: '1.5rem'}}>ID</TableCell>
                        <TableCell sx={{color: theme.typography.primary.color, fontSize: '1.5rem'}}>First name</TableCell>
                        <TableCell sx={{color: theme.typography.primary.color, fontSize: '1.5rem'}}>Last name</TableCell>
                        <TableCell sx={{color: theme.typography.primary.color, fontSize: '1.5rem'}}>Email</TableCell>
                        <TableCell sx={{color: theme.typography.primary.color, fontSize: '1.5rem'}}>Status</TableCell>
                    </TableRow>
                </TableHead>
                <TableBody>                    
                    {users.map((user, i) =>
                    <TableRow key={i}>                        
                        <TableCell sx={{color: theme.typography.primary.color}}>{user.id}</TableCell>
                        <TableCell sx={{color: theme.typography.primary.color}}>{user.firstName}</TableCell>
                        <TableCell sx={{color: theme.typography.primary.color}}>{user.lastName}</TableCell>
                        <TableCell sx={{color: theme.typography.primary.color}}>{user.email}</TableCell>
                        <TableCell sx={{color: theme.typography.primary.color}}>{user.state}</TableCell>
                    </TableRow>)}
                </TableBody>
            </Table>
        </Box>
    )
}