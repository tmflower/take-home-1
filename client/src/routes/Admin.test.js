import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Admin } from './Admin';

const users = [
    { "id": 1, "email": "kyle@getyodlr.com", "firstName": "Kyle", "lastName": "White", "state": "active"},
    { "id": 2, "email": "jane@getyodlr.com", "firstName": "Jane", "lastName": "Stone", "state": "active"},
    { "id": 3, "email": "lilly@getyodlr.com", "firstName": "Lilly", "lastName": "Smith", "state": "pending"},
    { "id": 4, "email": "fred@getyodlr.com", "firstName": "Fred", "lastName": "Miles", "state": "pending"},
    { "id": 5, "email": "alex@getyodlr.com", "firstName": "Alexandra", "lastName": "Betts", "state": "pending"}
    ]

test('renders Home without crashing', () => {
    render(<MemoryRouter>
            <Admin users={users}/>
        </MemoryRouter>);
});

test('matches snapshot', () => {
    const { asFragment } =
    render(
        <MemoryRouter>
            <Admin users={users}/>
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

test('renders Admin correctly', () => {
    render(
        <MemoryRouter>
            <Admin users={users}/>
        </MemoryRouter>
    );
    const h1 = screen.getByText('Admin Portal');
    expect(h1).toBeInTheDocument();
});