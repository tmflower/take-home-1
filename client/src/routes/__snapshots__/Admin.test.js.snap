// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-19btamt"
  >
    <h2
      class="MuiTypography-root MuiTypography-h2 css-5eqw5t-MuiTypography-root"
    >
      Admin Portal
    </h2>
    <table
      class="MuiTable-root css-rqglhn-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-375cik-MuiTableCell-root"
            scope="col"
          >
            ID
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-375cik-MuiTableCell-root"
            scope="col"
          >
            First name
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-375cik-MuiTableCell-root"
            scope="col"
          >
            Last name
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-375cik-MuiTableCell-root"
            scope="col"
          >
            Email
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-375cik-MuiTableCell-root"
            scope="col"
          >
            Status
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-5mac9p-MuiTableCell-root"
          >
            1
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-5mac9p-MuiTableCell-root"
          >
            Kyle
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-5mac9p-MuiTableCell-root"
          >
            White
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-5mac9p-MuiTableCell-root"
          >
            kyle@getyodlr.com
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-5mac9p-MuiTableCell-root"
          >
            active
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-5mac9p-MuiTableCell-root"
          >
            2
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-5mac9p-MuiTableCell-root"
          >
            Jane
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-5mac9p-MuiTableCell-root"
          >
            Stone
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-5mac9p-MuiTableCell-root"
          >
            jane@getyodlr.com
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-5mac9p-MuiTableCell-root"
          >
            active
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-5mac9p-MuiTableCell-root"
          >
            3
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-5mac9p-MuiTableCell-root"
          >
            Lilly
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-5mac9p-MuiTableCell-root"
          >
            Smith
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-5mac9p-MuiTableCell-root"
          >
            lilly@getyodlr.com
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-5mac9p-MuiTableCell-root"
          >
            pending
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-5mac9p-MuiTableCell-root"
          >
            4
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-5mac9p-MuiTableCell-root"
          >
            Fred
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-5mac9p-MuiTableCell-root"
          >
            Miles
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-5mac9p-MuiTableCell-root"
          >
            fred@getyodlr.com
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-5mac9p-MuiTableCell-root"
          >
            pending
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-5mac9p-MuiTableCell-root"
          >
            5
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-5mac9p-MuiTableCell-root"
          >
            Alexandra
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-5mac9p-MuiTableCell-root"
          >
            Betts
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-5mac9p-MuiTableCell-root"
          >
            alex@getyodlr.com
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-5mac9p-MuiTableCell-root"
          >
            pending
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
